batch-submitter/drivers/proposer/driver.go

```diff
@@ -264,19 +264,20 @@ func (d *Driver) CraftBatchTx(
 		d.lastStart = start
 		d.lastCommitTime = time.Now().Add(-d.cfg.PollInterval)
 	}
-
 	//If the waiting time has not been reached, then check whether the minimum stateroot number
 	//is met. if not, return nil
-	if time.Now().Add(-d.cfg.MaxBatchSubmissionTime).Before(d.lastCommitTime) {
-		// Abort if we don't have enough state roots to meet our minimum
-		// requirement.
-		rangeLen := end.Uint64() - start.Uint64()
-		if rangeLen < d.cfg.MinStateRootElements {
+	rollupTxn := end.Uint64() - start.Uint64()
+	if rollupTxn < d.cfg.MinStateRootElements && (time.Now().Add(-d.cfg.RollupTimeout).Before(d.lastCommitTime) || rollupTxn < d.cfg.MinTimeoutStateRootElements) {
```

Comment:
Check the size of rollupTxn and MinTimeoutStateRootElements only after the timeout has occurred.